SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java

```diff
@@ -368,16 +382,27 @@ CheckTaskState getTaskState(SingularityTask task, Optional<SingularityRequestWit
     return Optional.absent();
   }
 
-  private boolean isOverdue(SingularityTask task) {
-    final long taskDuration = System.currentTimeMillis() - task.getTaskId().getStartedAt();
+  private boolean isHealthcheckOverdue(SingularityTask task) {
+    final long healthcheckDuration = taskManager.getLastHealthcheck(task.getTaskId()).get().getDurationMillis().or(0L);
```

Comment:
I wasn't sure of the best route here since we don't want to use the time the task was launched for health checks. If we reach this logic, that means there is a health check so we can for sure have that, but the duration millis can be absent. If the duration never gets updated, this __could__ lead to the case of a health check running forever